src/lib.rs

```diff
@@ -82,6 +84,9 @@ impl fmt::Display for ActorError {
             ActorError::ChannelDisconnected { actor_name } => {
                 write!(f, "The message channel is disconnected for the actor {}.", actor_name)
             },
+            ActorError::ControlChannelDisconnected { actor_name } => {
```

Comment:
nit: `ControlChannel` is an internal concept, that the client of the library does not need to know about. You think it's okay to expose it as part of public API? or should we return `SystemStopped` since it's the only possible cause of `ControlChannelDisconnected`?

Reply:
I think that makes sense :) `unreachable!()/expect()` would be more self-documenting.

Reply:
Pushed the change. It resulted in more code cleanups, so I'll wait for another round of review before merging.